### PR TITLE
implemented 11-get-articles-by-query

### PR DESCRIPTION
**Pull request by DaniMaron**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -370,12 +378,59 @@ describe("GET /api/users", () => {
...
+      });
+  });
+  test("Status 404: responds with an appropriate error when query value does not produce any article", () => {
+    return request(app)
+      .get("/api/articles?topic=73")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("Not found");
+      });
+  });
```
Comment (Harrison365) [inline __tests__/app.test.js:433]: Very good 👍 
Just one more test needed: Please make a request for a topic which exists (200) but that has no articles. This should return an empty array.

File: models/articles.model.js [hunk 2]
```diff
@@ -11,16 +11,44 @@ function fetchArticleById(id) {
...
+
+  const queriesKeys = Object.keys(queries);
+  if (queriesKeys.length > 0) {
+    for (let i = 0; i < queriesKeys.length; i++) {
+      if (acceptableFilteringQueries.includes(queriesKeys[i])) {
+        if (i === 0) dbQueryMiddle += ` WHERE articles.${queriesKeys[i]} = $1`;
+        else
+          dbQueryMiddle += `
+         AND articles.${queriesKeys[i]} = $2`;
+        values.push(queries[queriesKeys[i]]);
```
Comment (Harrison365) [inline models/articles.model.js:43]: Cool 👍

File: models/articles.model.js [hunk 3]
```diff
@@ -11,16 +11,44 @@ function fetchArticleById(id) {
...
+      } else {
+        dbQueryMiddle += ` WHERE articles.kdsljs = 'mitch'`;
+      }
```
Comment (Harrison365) [inline models/articles.model.js:46]: Sorry, I don't understand this bit 😕 ^^^

Comment (DaniMaron) [reply models/articles.model.js:46]: I basically wanted to get an error when the column from the query doesn't exist, but without risking injection by plugging whatever the user wrote as a column. So I went for a fake word that I'm sure is safe.

**Pull request merged**